src/test/java/io/vertx/tests/interceptors/BodyInterceptorTest.java

```diff
@@ -56,15 +58,16 @@ public void testModifyRequestJson(TestContext ctx) {
       }))));
 
     String content = "{\"k2\": 2}";
-    vertx.createHttpClient().request(HttpMethod.POST, 8080, "localhost", "/")
+    HttpClient client = vertx.createHttpClient();
+    client.request(HttpMethod.POST, 8080, "localhost", "/")
       .compose(request -> request
         .putHeader("Content-Type", "application/json")
         .putHeader("Content-Length", "" + content.length())
         .write(content)
         .compose(r -> request.send())
       )
       .onComplete(ctx.asyncAssertSuccess(response -> {
-        latch.complete();
+        client.close().onSuccess(x -> latch.complete());
```

Comment:
It's fine to create the client when it is necessary, but it's a better practice to close in a `tearDown` method (will be executed regardless of the test result)

Reply:
Do you mean by still only creating the client at `startUp` and only closing it at `tearDown` ?

Reply:
Something like this. But you simply invoke `close()` if the client is not null and then don't forget to invoke `super.tearDown(context)`